### DISCUSSION TITLE
What does s_priceFeed do that priceFeed doesn't?

### DISCUSSION THREAD

**Discussion by mxnwafor** [Cyfrin/foundry-full-course-cu#1540] [Q&A] [answered]
Hey guys, I'm having an issue understanding the reason behind creating the `s_priceFeed` variable and not using the `priceFeed` variable passed into the constructor. Why do we make it of type `AggregatorV3Interface`?

```
address public /* immutable */ i_owner;
    uint256 public constant MINIMUM_USD = 5 * 10 ** 18;
    AggregatorV3Interface private s_priceFeed;
    
    constructor(address priceFeed) { // you'll pass the address of the chain/token you wish to work with
        i_owner = msg.sender;
        s_priceFeed = AggregatorV3Interface(priceFeed);
    }

    function fund() public payable {
        require(msg.value.getConversionRate(s_priceFeed) >= MINIMUM_USD, "You need to spend more ETH!");
        // require(PriceConverter.getConversionRate(msg.value) >= MINIMUM_USD, "You need to spend more ETH!");
        addressToAmountFunded[msg.sender] += msg.value;
        funders.push(msg.sender);
```

Please, any clear explanations would be highly appreciated <3

**Comment by EngrPips** [comment, marked as answer]
Hello @0xMitchy  For us to get the price of ETH/USD we need to call a contract of type  `AggregatorV3Interface` that have been deployed to some address on the blockchain, so in our contract we created a type of `AggregatorV3Interface` named `s_priceFeed` and then we went to chainlink to get an address for a contract of type `AggregatorV3Interface` which the sole responsibility is to provide the current price of ETH/USD for whoever needs it and that contract was deployed to an address which is the `priceFeed` address that we pass into the constructor.

We can't call a function on an address but we can only call function on a contract, hence that is why we wrapped the `priceFeed` address that we passed in to the constructor with the `AggregatorV3Interface` interface to form a contract type of `AggregatorV3` which was deployed to the `priceFeed`  address.

So if you check the PriceConverter library you would see that we call a method on the `s_priceFeed` because we have made a contract type that have an interface of `AggregatorV3Interface` [from this interface we would get all the functions we want to call] and the contract is deployed to `priceFeed` address, so whenever we calling a method on `s_priceFeed` we are sending a transaction to the `priceFeed` address on the blockchain.

I really hope i did well to explain this, please let me know if anything still seem confusing to you.

**Reply by mxnwafor** [reply]
Thank you so much @EngrPips 
I fully understand it now; after going through the video and the code a couple more times, then reading this, it all makes sense.

The `priceFeed` is of type `address`, so it doesn't really do much unless it is of type `AggregatorV3Interface`, which is now `s_priceFeed`. This `s_priceFeed` now has the functionalities and ability to grab the ETH/USD price from the interface.

So basically, `s_priceFeed` is `priceFeed` but with a functionality that helps get the price and interact with other embedded methods.

Thanks once again.

**Reply by EngrPips** [reply]
Awesome. I am really glad you understand the concept now. Honestly for anyone to understand anything it first of all have to make sense to the person. So whenever you want to understand make sure to free yourself from the attachment of whatever definition the tutor or any random person gives and try to make your own sense out of it within the context of what the stuff is.
